openpype/hosts/maya/api/plugin.py

```diff
@@ -187,7 +221,7 @@ def load(
 
         return loaded_containers
 
-    def process_reference(self, context, name, namespace, data):
+    def process_reference(self, context, name, namespace, data, group_name=None):  # noqa
```

Comment:
I wanted to stress again that this could be a backwards incompatible change for any studio (or client?) which might have developed additional Loader plug-ins for Maya which inherit from this class because those will need to be refactored to also take this additional arguments.

A potential backwards compatible way would have been to maybe put the group name into the `data` dictionary as additional entry. Also, `data` here should be renamed to `options`. It was wrong before apparently but _luckily?_ was overridden on each inheriting plug-in to be `options` instead. With the argument being called `data` the call on line 200 (before line 167) wouldn't work because it explicitly passes `options` as a keyword.

The backwards compatible `_load_animation` plug-in would need to also be refactored `data` -> `options`. With it being broken currently I have a feeling that the old load animation hasn't been in use for quite some time otherwise we'd have seen this reported maybe?

Reply:
> A potential backwards compatible way would have been to maybe put the group name into the data dictionary as additional entry. Also, data here should be renamed to options. It was wrong before apparently but luckily? was overridden on each inheriting plug-in to be options instead. With the argument being called data the call on line 200 (before line 167) wouldn't work because it explicitly passes options as a keyword.

Yup good point. Lets do that.

> The backwards compatible _load_animation plug-in would need to also be refactored data -> options. With it being broken currently I have a feeling that the old load animation hasn't been in use for quite some time otherwise we'd have seen this reported maybe?

Pretty sure any Pyblish plugins starting with `_` will be ignored, so it should actually be removed.

Reply:
I think it might be there to allow 'managing' any loaded content which had previously been loaded using that loader - but not sure. :)

Reply:
@BigRoy So if I understand correctly, you want to rename the argument `data` to `options` in the different `process_reference` methods?
And also pass the `group_name` as a new entry in this `options` so we don't need to refactor all classes that inherit this one? But the refactoring is inevitable because even if we pass the `group_name` as new entry, we need to modify all chidren classes to get the group name from the options, don't we?

Reply:
> So if I understand correctly, you want to rename the argument data to options in the different process_reference methods?

Yes. This we should do either way since it's basically been invalid before.

> And also pass the group_name as a new entry in this options so we don't need to refactor all classes that inherit this one?

Yes. This is what I'm proposing. Whether we should do that maybe @tokejepsen and @antirotor have an opinion on the matter.

> But the refactoring is inevitable because even if we pass the group_name as new entry, we need to modify all chidren classes to get the group name from the options, don't we?

Sort of. Without refactoring it would not use the predefined group name, but at the same time it also wouldn't error. It would then at least not break existing custoom plugins someone might have in their studio.

Reply:
> Yes. This is what I'm proposing. Whether we should do that maybe @tokejepsen and @antirotor have an opinion on the matter.

@BigRoy Yeah, I think this is the best route to take. Wasnt keen on the idea of adding the extra keyword argument and its a good point about external people might be using it.

> But the refactoring is inevitable because even if we pass the group_name as new entry, we need to modify all chidren classes to get the group name from the options, don't we?

@friquette True we'll need to refactor anyways for this feature, but studios wont need to refactor if they use the method. They'll need to refactor if they want to use this feature, but that is expected for new features.